src/library/scala/collection/mutable/FlatHashTable.scala

```diff
@@ -219,7 +219,9 @@ trait FlatHashTable[A] extends FlatHashTable.HashUtils[A] {
     tableSize = 0
     nnSizeMapReset(table.length)
     seedvalue = tableSizeSeed
-    threshold = newThreshold(_loadFactor, table.length)
+    // If the table length reaches MaximumTableCapacity of Int.MaxValue / 2 then set the
+    // threshold to Int.MaxValue - 8 as a safe value
```

Comment:
Why is this "safe"?  If you never try to resize your table when getting near the maximum size, your performance will get worse and worse and worse as you add elements and eventually terminate in an infinite loop when there are no more free slots available in the flat table.